### ISSUE TITLE
Switch AI can select same mon twice in double battles

### ISSUE THREAD

**Issue by Pawkkie** [rh-hideout/pokeemerald-expansion#5451]
### Description

Found this when working on a refactor to enable switch prediction. Egg wrote a fix for this in #4098, but it turns out that fix only applies inside the `AI_ShouldSwitchIfBadMoves` function, which only comes up for switches where the AI has no good moves to use. All of the other types of switches may still have the bug present.

I'll PR a refactor for `AI_ShouldSwitchIfBadMoves` in a second, but it doesn't have this fully fixed as it's a lot of changes and I wanted to split it up to be easier to review. The refactor fixes this bug in *most* cases. Fixing it in *every* case will be a follow up PR once the refactor is merged. It all depends on the refactor so I'm gonna put it to upcoming.

EDIT: Should now be entirely fixed by #5452 

### Version

1.9.2 (Latest release)

**Comment by Pawkkie** [comment]
Closed by #5452
